File: Thiwanka_Sir_Algo/Naive_Algorithm.java
public class Main {

    static void search(String txt, String pat) {
        int l1 = txt.length();
        int l2 = pat.length();
        int i, j;

        for (i = 0; i <= l1 - l2; i++) {
            j = 0;
            while (j < l2 && txt.charAt(i + j) == pat.charAt(j)) {
                j++;
            }
            if (j == l2) {
                System.out.println("Pattern found at index: " + i);
            }
        }
    }

    public static void main(String[] args) {
        String txt = "ABAACAADAABAAABAA";
        String pat = "AABA";
        search(txt, pat);
    }
}

File: Thiwanka_Sir_Algo/Naive_algorithm.cpp
#include<iostream>
using namespace std;

void search(string array, string pattern) {
    if (array.length() >= pattern.length()) {
        for (int i = 0; i <= array.length() - pattern.length(); i++) {
            bool found = true;
            for (int j = 0; j < pattern.length(); j++) {
                if (array[i + j] != pattern[j]) {
                    found = false;
                    break;
                }
            }
            if (found) {
                cout << "Pattern found at index " << i << endl;
            }
        }
    } else {
        cout << "Array length is less than pattern length!" << endl;
    }
}

int main() {
    string array = "AABAfyuhjbAABAdrctygvhjAABA";
    string pattern = "AABA";

    search(array, pattern);

    return 0;
}
